### PR TITLE
fix AttributeSet_Set_Allow_Null read uninitialized memory

### PR DESCRIPTION
**Pull request by AviAvni**

### CODE REVIEW

File: src/graph/entities/attribute_set.c [hunk 1]
```diff
@@ -212,10 +212,14 @@ void AttributeSet_Set_Allow_Null
 		return;
 	}
 
-	size_t n = ATTRIBUTESET_BYTE_SIZE(_set) + sizeof(Attribute);
-	_set = rm_realloc(_set, n);
-
 	// allocate room for new attribute
+	size_t n = ATTRIBUTESET_BYTE_SIZE(_set) + sizeof(Attribute);
+	if(set == NULL) {
```
Comment (swilly22) [inline src/graph/entities/attribute_set.c:217]: this condition is in conflict with the ASSERT above, should we remove that assertion ?

File: src/graph/entities/attribute_set.c [hunk 2]
```diff
@@ -212,10 +212,14 @@ void AttributeSet_Set_Allow_Null
 		return;
 	}
 
-	size_t n = ATTRIBUTESET_BYTE_SIZE(_set) + sizeof(Attribute);
```
Comment (swilly22) [inline src/graph/entities/attribute_set.c:215]: How is this function (`AttributeSet_Set_Allow_Null`) different from `AttributeSet_Update`, meaning if we made a change here do we need to also apply a similar change to `AttributeSet_Update` ?

File: src/graph/entities/attribute_set.c [hunk 3]
```diff
@@ -212,10 +212,14 @@ void AttributeSet_Set_Allow_Null
...
-	_set = rm_realloc(_set, n);
```
Comment (swilly22) [inline src/graph/entities/attribute_set.c:216]: So the issue with `rm_realloc` is that in-case `_set` is NULL `rm_realloc` won't zero out the allocated space ?

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: src/graph/entities/attribute_set.c (hunk 3 above)
Comment (AviAvni) [reply src/graph/entities/attribute_set.c:216]: Yes

File: src/graph/entities/attribute_set.c (hunk 1 above)
Comment (AviAvni) [reply src/graph/entities/attribute_set.c]: fixed

File: src/graph/entities/attribute_set.c (hunk 2 above)
Comment (AviAvni) [reply src/graph/entities/attribute_set.c:215]: not `AttributeSet_Update` `AttributeSet_Add` and now they are similar

**Pull request merged**